File: leetcode/go/26.remove-duplicates-from-sorted-array.go
/*
 * @lc app=leetcode id=26 lang=golang
 *
 * [26] Remove Duplicates from Sorted Array
 */

// @lc code=start
func removeDuplicates(nums []int) int {
	length := len(nums)
	if length == 0 {
		return 0
	}
	count := 1
	for i := 0; i+1 < length; i++ {
		if nums[i] == nums[i + 1] {
			continue
		} else {
			nums[count] = nums[i + 1]
			count++
		}
	}
	return count
}

// @lc code=end

File: leetcode/cpp/26.remove-duplicates-from-sorted-array.cpp
/*
 * @lc app=leetcode id=26 lang=cpp
 *
 * [26] Remove Duplicates from Sorted Array
 */
#include <iostream>
#include <vector>
using namespace std;
// @lc code=start
class Solution {
public:
    int removeDuplicates(vector<int>& nums) {
        int pos = 0;
        for(int i = 1; i < nums.size(); ++i){
            if(nums[i] != nums[pos]){
                ++pos;
                nums[pos] = nums[i];
            }
        }
        return pos + 1;
    }
};
// @lc code=end

